cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/submit/KubeSparkSubmitter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.app.runtime.spark.submit;
+
+import com.google.common.collect.ImmutableList;
+import io.cdap.cdap.app.runtime.spark.SparkRuntimeContext;
+import io.cdap.cdap.app.runtime.spark.SparkRuntimeEnv;
+import io.cdap.cdap.app.runtime.spark.SparkRuntimeUtils;
+import io.cdap.cdap.app.runtime.spark.distributed.SparkExecutionService;
+import io.cdap.cdap.internal.app.runtime.workflow.BasicWorkflowToken;
+import io.cdap.cdap.internal.app.runtime.workflow.WorkflowProgramInfo;
+import io.cdap.cdap.master.spi.environment.spark.SparkConfig;
+import io.cdap.cdap.proto.id.ProgramRunId;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.yarn.api.ApplicationConstants;
+import org.apache.twill.filesystem.LocationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Kubernetes spark submitter.
+ */
+public class KubeSparkSubmitter extends AbstractSparkSubmitter {
+  private static final Logger LOG = LoggerFactory.getLogger(KubeSparkSubmitter.class);
+
+  private final SparkExecutionService sparkExecutionService;
+  private final SparkConfig sparkConfig;
+
+  /**
+   * Kube spark submitter constructor.
+   */
+  public KubeSparkSubmitter(LocationFactory locationFactory, String hostname,
+                            SparkRuntimeContext runtimeContext, SparkConfig sparkConfig) {
+    ProgramRunId programRunId = runtimeContext.getProgram().getId().run(runtimeContext.getRunId().getId());
+    WorkflowProgramInfo workflowInfo = runtimeContext.getWorkflowInfo();
+    BasicWorkflowToken workflowToken = workflowInfo == null ? null : workflowInfo.getWorkflowToken();
+    this.sparkExecutionService = new SparkExecutionService(locationFactory, hostname, programRunId, workflowToken);
+    this.sparkConfig = sparkConfig;
+  }
+
+  @Override
+  protected Map<String, String> getSubmitConf() {
+    Map<String, String> config = new HashMap<>();
+    config.put("spark.kubernetes.driverEnv.CDAP_LOG_DIR", ApplicationConstants.LOG_DIR_EXPANSION_VAR);
```

Comment:
anything kubernetes specific should come from the MasterEnvironment and not from this class.